rts/motoko-rts-tests/src/gc.rs

```diff
@@ -222,6 +254,26 @@ fn compute_reachable_objects(
     closure
 }
 
+fn check_closure_table(mut offset: usize, closure_table: &[ObjectIdx], heap: &[u8]) {
+    assert_eq!(read_word(heap, offset), TAG_ARRAY);
+    offset += WORD_SIZE;
+
+    assert_eq!(read_word(heap, offset), closure_table.len() as u32);
+    offset += WORD_SIZE;
+
+    for obj in closure_table.iter() {
+        let ptr = unskew_pointer(read_word(heap, offset));
+        offset += WORD_SIZE;
+
+        // Skip object header for idx
+        let idx_address = ptr as usize + size_of::<Array>().to_bytes().0 as usize;
+        let idx = get_scalar_value(read_word(heap, idx_address - heap.as_ptr() as usize));
```

Comment:
I don't understand what this line is doing, especially `idx_address - heap.as_ptr`?

Reply:
That is computing offset of index field of the object pointed by the table in `heap`. Pointers in heap are addresses, we need to convert those to offsets in `heap` to be able to access via `heap[...]`.

Reply:
Gotcha.